src/Model/ModelTrait.php

```diff
@@ -140,13 +145,16 @@ private function getFirstParameter($method)
      * @param \ReflectionParameter $param
      * @param mixed $value
      * @param bool $isCollection
-     * @return object
+     * @return mixed
+     * @throws ModelException
+     * @throws \ReflectionException
      */
-    private function constructClasses(\ReflectionParameter $param, $value, $isCollection = false)
+    private function constructClasses(\ReflectionParameter $param, mixed $value, bool $isCollection = false): mixed
     {
-        $class = $param->getClass();
-        if (!empty($class)) {
+        $class = $param->getType() && !$param->getType()->isBuiltin()
+            ? new \ReflectionClass($param->getType()->getName()) : null;
 
+        if (!empty($class)) {
```

Comment:
Think I'd prefer to check if it's an instance of `ReflectionClass` rather than empty, so we're explicit about what we expect here

Reply:
Fixed